datastorages/jcloud/src/main/java/org/fao/geonet/api/records/attachments/JCloudStore.java

```diff
@@ -136,28 +147,55 @@ private MetadataResource createResourceDescription(final ServiceContext context,
-            versionValue = storageMetadata.getETag(); // ETAG is cryptic may need some other value?
+            String versionPropertyName = jCloudConfiguration.getExternalResourceManagementVersionPropertyName();
+            if (StringUtils.hasLength(versionPropertyName)) {
+                if (storageMetadata.getUserMetadata().containsKey(versionPropertyName)) {
+                    versionValue = storageMetadata.getUserMetadata().get(versionPropertyName);
+                } else {
+                    versionValue = "";
+                    Log.warning(Geonet.RESOURCES, String.format("Expecting property '%s' on resource '%s' for metadata %d(%s) but the property was not found.",
+                        versionPropertyName, resourceId, metadataId, metadataUuid));
+                }
+            } else {
+                versionValue = storageMetadata.getETag();
+            }
         }
 
         MetadataResourceExternalManagementProperties.ValidationStatus validationStatus = MetadataResourceExternalManagementProperties.ValidationStatus.UNKNOWN;
-        if (!StringUtils.isEmpty(jCloudConfiguration.getExternalResourceManagementValidationStatusPropertyName())) {
+        if (StringUtils.hasLength(jCloudConfiguration.getExternalResourceManagementValidationStatusPropertyName())) {
             String validationStatusPropertyName = jCloudConfiguration.getExternalResourceManagementValidationStatusPropertyName();
             String propertyValue = null;
             if (storageMetadata.getUserMetadata().containsKey(validationStatusPropertyName)) {
                 propertyValue = storageMetadata.getUserMetadata().get(validationStatusPropertyName);
             }
-            if (StringUtils.isNotEmpty(propertyValue)) {
+            if (StringUtils.hasLength(propertyValue)) {
                 validationStatus = MetadataResourceExternalManagementProperties.ValidationStatus.fromValue(Integer.parseInt(propertyValue));
             }
         }
 
         MetadataResourceExternalManagementProperties metadataResourceExternalManagementProperties =
-            getMetadataResourceExternalManagementProperties(context, metadataId, metadataUuid, visibility, resourceId, filename, storageMetadata.getETag(), storageMetadata.getType(), validationStatus);
+            getMetadataResourceExternalManagementProperties(context, metadataId, metadataUuid, visibility, resourceId, filename, storageMetadata.getETag(), storageMetadata.getType(),
+                validationStatus);
 
         return new FilesystemStoreResource(metadataUuid, metadataId, filename,
-            settingManager.getNodeURL() + "api/records/", visibility, storageMetadata.getSize(), storageMetadata.getLastModified(), versionValue, metadataResourceExternalManagementProperties, approved);
+            settingManager.getNodeURL() + "api/records/", visibility, storageMetadata.getSize(), changedDate, versionValue, metadataResourceExternalManagementProperties, approved);
```

Comment:
Just to check if `changedDate` or `versionValue` are `null`, if could cause issues in `FilesystemStoreResource`. In the previous code I have no clear if could be `null`.

Reply:
I fixed changeDate so that it will not be null like before.
For versionValues, it was previously null if versioning was not enabled so it should not require any change.